src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlDataContextTypeTransformer.cs

```diff
@@ -24,7 +24,6 @@ public IXamlAstNode Transform(AstTransformationContext context, IXamlAstNode nod
             {
                 AvaloniaXamlIlDataContextTypeMetadataNode inferredDataContextTypeNode = null;
                 AvaloniaXamlIlDataContextTypeMetadataNode directiveDataContextTypeNode = null;
-                bool isDataTemplate = on.Type.GetClrType().Equals(context.GetAvaloniaTypes().DataTemplate);
```

Comment:
Can we update the assignment here instead of querying the type system twice?

Reply:
They're different, one is `IDataTemplate`, the other is `DataTemplate`.

Reply:
I missed that. This looks good to me then!